src/protocols/http/hpr_http_worker.erl

```diff
@@ -0,0 +1,259 @@
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+init(Args) ->
+    #{
+        protocol := #http_protocol{
+            endpoint = Address,
+            flow_type = FlowType,
+            dedupe_timeout = DedupeTimeout,
+            auth_header = Auth,
+            protocol_version = ProtocolVersion
+        },
+        net_id := NetID
+    } = Args,
+    lager:debug("~p init with ~p", [?MODULE, Args]),
+    {ok, #state{
+        net_id = NetID,
+        address = Address,
+        transaction_id = next_transaction_id(),
+        send_data_timer = DedupeTimeout,
+        flow_type = FlowType,
+        auth_header = Auth,
+        protocol_version = ProtocolVersion,
+        routing_info = undefined
+    }}.
+
+handle_call(_Msg, _From, State) ->
+    lager:warning("rcvd unknown call msg: ~p from: ~p", [_Msg, _From]),
+    {reply, ok, State}.
+
+handle_cast(
+    {handle_packet, PacketUp, GatewayTime, GatewayStream, RoutingInfo},
+    #state{send_data_timer = 0, shutdown_timer_ref = ShutdownTimerRef0} = State
+) ->
+    {ok, StateWithPacket} = do_handle_packet(
+        PacketUp, GatewayTime, GatewayStream, RoutingInfo, State
+    ),
+    ok = send_data(StateWithPacket),
+    {ok, ShutdownTimerRef1} = maybe_schedule_shutdown(ShutdownTimerRef0),
+    {noreply, State#state{shutdown_timer_ref = ShutdownTimerRef1}};
+handle_cast(
+    {handle_packet, PacketUp, GatewayTime, GatewayStream, RoutingInfo},
+    #state{
+        should_shutdown = false,
+        send_data_timer = Timeout,
+        send_data_timer_ref = TimerRef0
+    } = State0
+) ->
+    {ok, State1} = do_handle_packet(PacketUp, GatewayTime, GatewayStream, RoutingInfo, State0),
+    {ok, TimerRef1} = maybe_schedule_send_data(Timeout, TimerRef0),
+    {noreply, State1#state{send_data_timer_ref = TimerRef1}};
+handle_cast(
+    {handle_packet, _PacketUp, _PacketTime, _GatewayStream, _RoutingInfo},
+    #state{
+        should_shutdown = true,
+        shutdown_timer_ref = ShutdownTimerRef0,
+        send_data_timer = DataTimeout
+    } = State0
+) ->
+    lager:info("packet delivery after data sent [send_data_timer: ~p]", [DataTimeout]),
+    {ok, ShutdownTimerRef1} = maybe_schedule_shutdown(ShutdownTimerRef0),
+    {noreply, State0#state{shutdown_timer_ref = ShutdownTimerRef1}};
+handle_cast(_Msg, State) ->
+    lager:warning("rcvd unknown cast msg: ~p", [_Msg]),
+    {noreply, State}.
+
+handle_info(?SEND_DATA, #state{} = State) ->
+    ok = send_data(State),
+    {ok, ShutdownTimerRef} = maybe_schedule_shutdown(undefined),
+    {noreply, State#state{should_shutdown = true, shutdown_timer_ref = ShutdownTimerRef}};
+handle_info(?SHUTDOWN, #state{} = State) ->
+    {stop, normal, State};
+handle_info(_Msg, State) ->
+    lager:warning("rcvd unknown info msg: ~p, ~p", [_Msg, State]),
+    {noreply, State}.
+
+code_change(_OldVsn, State, _Extra) ->
```

Comment:
This callback is now optional. It can be removed since it's just boilerplate here.

Reply:
Adios, boilerplate!